k8s/client.go

```diff
@@ -448,3 +456,206 @@ func (client *Client) FindDeploymentByName(ctx context.Context, name string) (*a
+	if err != nil {
+		return nil, err
+	}
+	return applied, nil
+
+}
+
+func applyOperatorManifest(ctx context.Context, config *restclient.Config, manifestFull []byte) ([][]string, error) {
+	dynamicClient, err := dynamic.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	manifests := splitManifest(manifestFull)
+
+	var appliedSuccessfully [][]string
+	for _, manifest := range manifests {
+		manifest = strings.TrimSpace(manifest)
+		if manifest == "" {
+			continue
+		}
+
+		decoder := yamlk8s.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
+		obj := &unstructured.Unstructured{}
+		_, gvk, err := decoder.Decode([]byte(manifest), nil, obj)
+		if err != nil {
+			fmt.Printf("Failed to decode manifest: %v", err)
+			//TODO: implement rollback, maybe one level up
+			//err := rollbackManifests(ctx, config, appliedSuccessfully)
+			return nil, err
+		}
+
+		kindPluralized := strings.ToLower(gvk.Kind) + "s"
+		resource := dynamicClient.Resource(gvk.GroupVersion().WithResource(kindPluralized))
+
+		created, err := resource.Namespace(obj.GetNamespace()).Create(ctx, obj, metav1.CreateOptions{})
+		if err != nil {
+			fmt.Printf("Failed to apply manifest: %v", err)
+			return nil, err
+		}
+
+		appliedSuccessfully = append(appliedSuccessfully, []string{created.GetKind(), created.GetName()})
+	}
+	return appliedSuccessfully, nil
+}
+
+//func rollbackManifests(ctx context.Context, config *restclient.Config, manifests [][]string) error {
+//	if len(manifests) == 0 {
+//		return nil
+//	}
+//	fmt.Printf("An error ocurred while applying manifests\n")
+//	fmt.Printf("Trying to rollback %d manifests\n", len(manifests))
+//
+//	dynamicClient, err := dynamic.NewForConfig(config)
+//	if err != nil {
+//		return err
+//	}
+//	fmt.Printf("Trying to rollback %d manifests\n", len(manifests))
+//	for _, manifest := range manifests {
+//		manifest = strings.TrimSpace(manifest)
+//		if manifest == "" {
+//			continue
+//		}
+//
+//		decoder := yamlk8s.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
+//		obj := &unstructured.Unstructured{}
+//		_, gvk, err := decoder.Decode([]byte(manifest), nil, obj)
+//		if err != nil {
+//			fmt.Printf("Failed to decode manifest: %v", err)
+//			continue
+//		}
+//
+//		kindPluralized := strings.ToLower(gvk.Kind) + "s"
+//		resource := dynamicClient.Resource(gvk.GroupVersion().WithResource(kindPluralized))
+//
+//		err = resource.Namespace(obj.GetNamespace()).Delete(ctx, obj.GetName(), metav1.DeleteOptions{})
+//		if err != nil {
+//			fmt.Printf("Failed to delete manifest: %v", err)
+//			return err
+//		}
+//
+//		fmt.Printf("Deleted %s %s\n", obj.GetKind(), obj.GetName())
+//	}
+//	return nil
+//}
+
+func splitManifest(manifest []byte) []string {
+	return strings.Split(string(manifest), "---\n")
+}
+
+func getOperatorManifest(version string) ([]byte, error) {
+	const operatorReleases = "https://api.github.com/repos/calyptia/core-operator-releases/releases"
```

Comment:
Make those constant a first level. Not under a function

Reply:
i set under the function because is not needed anywhere else